Implement evolution chain

piotr-wawro opened the pull request:
<img src="https://user-images.githubusercontent.com/44544007/119706542-670a6280-be5a-11eb-9fe1-e8d9c6382448.gif" width="300"/>

piotr-wawro committed:
Implement evolution chain

piotr-wawro committed:
Add information about evolution requirements

RANDOM-V2N5q commented:
<img src="https://user-images.githubusercontent.com/44544007/119719731-6deca180-be69-11eb-8804-db28cb5d50d2.gif" width="300" />

piotr-wawro committed:
Capitalize evolution items namse

piotr-wawro committed:
Fix typo

piotr-wawro committed:
Fix arrow not displaying for second evolution

piotr-wawro committed:
Fix pokemon colors on evolution card

kamil-matula committed:
Refactor Evolution

DONE:
1. Added new attribute in themes.js for evolution card background.
2. Moved capitalizing text from EvolutionArrow to Evolution, so only item names are capitalized.
3. Added missing padding to Moves List Header.
4. Refactored Evolution to avoid repeating the code.

kamil-matula commented on src/screens/details/Evolution.js:
```diff
@@ -1,17 +1,69 @@
 
+  var tier1 = chain;
+  var tier2 = chain.evolvesTo;
+  var tier3;
+
+  if (tier2.length == 1) {
+    tier3 = tier2[0].evolvesTo;
+  } else {
+    tier3 = [];
+  }
```
You can write it in one line: `var tier3 = tier2.length == 1 ? tier2[0].evolvesTo : [];`

kamil-matula reviewed (approved):
It looks really pretty and works perfectly, so good job!

kamil-matula committed:
Shorten variable initialization

Blazevarjo commented on src/screens/details/Evolution.js:
```diff
@@ -1,17 +1,63 @@
+import { SafeAreaView, StyleSheet, Dimensions, View, ScrollView } from 'react-native';
 import { useTheme } from 'react-native-paper';
 
 import { DetailsAppBar } from '../../components';
+import EvolutionArrow from '../../components/EvolutionArrow';
+import EvolutionCard from '../../components/EvolutionCard';
```
``
import { DetailsAppBar, EvolutionArrow, EvolutionCard  } from '../../components';
``

Blazevarjo commented on src/screens/details/Evolution.js:
```diff
@@ -1,17 +1,63 @@
 export default function Evolution({ color, sprite, chain }) {
   const colors = useTheme().colors;
 
+  var tier1 = chain;
+  var tier2 = chain.evolvesTo;
+  var tier3 = tier2.length == 1 ? tier2[0].evolvesTo : [];
```
var => const

Blazevarjo reviewed (commented):
Everything seems fine, only slight refactor. gj
